builtin/mainmenu/tab_about.lua

```diff
@@ -119,13 +119,44 @@ local function build_hacky_list(items, spacing)
 	return table.concat(ret, ""), y
 end
 
+local function read_text_file(path)
+	local f = io.open(path, "r")
+	if not f then
+		return nil
+	end
+    local text = f:read("*all")
+    f:close()
+	return text
+end
+
+local function collect_debug_info()
+	local version = core.get_version()
+	local irrlicht_device = core.get_active_irrlicht_device()
+	local renderer = core.get_active_renderer()
+
+	local path_prefix = core.get_user_path() .. DIR_DELIM
+	local minetest_conf = read_text_file(path_prefix .. "minetest.conf") or "[not available]\n"
+	local debug_txt = read_text_file(path_prefix .. "debug.txt") or "[not available]\n"
+
+	return table.concat({
+		version.project, " ", (version.hash or version.string), "\n",
+		"Platform: ", PLATFORM, "\n",
```

Comment:
Using

```c++
RenderingEngine::get_raw_device()->getOSOperator()->getOperatingSystemVersion()
```

Gives you:

```
Linux 6.4.12-arch1-1 #1 SMP PREEMPT_DYNAMIC Thu, 24 Aug 2023 00:38:14 +0000 x86_64
```

Distro and X11 vs Wayland would be nice as well

Reply:
`IOSOperator::getOperatingSystemVersion` doesn't seem to be that great. Not sure if it returns anything useful on non-Linux platforms. With the SDL device, it only returns the SDL version.